### PR TITLE
Group Plans quick wins

### LINKED ISSUE #11109: claimed Group Plan task does not appear in personal task list until a sync
As reported by Lord Kraken @mbianchi (071314b5-b684-43da-8e19-118e36ea16f6) and confirmed by me:

_"When I make a Group Task (in this case it was a To Do, but I think it applies to any kind of Task) and assign it to myself, and then click the "Tasks" button on the Nav bar, the Task I just created and assigned to myself does not show up in my list. I then Refresh my browser window with F5 and the task shows up. i am using Chrome on Windows 10."_

The task will also appear if you click the sync button but ideally it shouldn't be necessary to do either a sync or a refresh.

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Makes a few updates to Group Plans for general UX improvement and minor code cleanup:

* Removes an unused `isManager` function from `group-plans/index.vue`
* Updates the task creation control on the group plans page to current foldout button style
* Shows/hides the task creation control based on whether the user is allowed to create group tasks
* Removes raw JavaScript confirmation dialog when approving group tasks
- [x] TO DO: Address how this messes up the UX of the task-needs-approval notification
* Disables task dragging on the Group Plan task board
- [ ] TO DO?: If possible, make task dragging actually work instead. It updates `tasksOrder` on the back end, but we don't make use of the order at the moment.
* Reenables the "..." options button on Group Plan tasks, for quick access to the delete button
* Fixes an API issue where deleting Group Plan tasks did not remove the corresponding `tasksOrder` entry
* Adds a sync when the user claims or unclaims a task, to fix #11109

### CODE REVIEW

File: website/server/models/group.js [hunk 1]
```diff
@@ -1418,6 +1418,10 @@ schema.methods.removeTask = async function groupRemoveTask (task) {
   }, {
     $set: {'group.broken': 'TASK_DELETED'},
   }, {multi: true}).exec();
+
+  removeFromArray(group.tasksOrder[`${task.type}s`], task._id);
```
Comment (paglias) [inline website/server/models/group.js:1422]: good catch!

**Comment by paglias** [comment]
Regarding

> Updates the task creation control on the group plans page to current foldout button style

can we reuse the css from the existing  button for the user?

**Comment by SabreCat** [comment]
@paglias Refactored in 3af1fb1, that look OK?

**Review by paglias — APPROVED** [review]
LGTM!

**Comment by fornsss** [comment]
pulpul

W dniu niedziela, 14 kwietnia 2019 Matteo Pagliazzi <
notifications@github.com> napisał(a):

**Comment by fornsss** [comment]
hej misiaki

pon., 15.04.2019, 17:48 użytkownik Sabe Jones <notifications@github.com>
napisał:

**Pull request merged**